core/src/main/java/io/questdb/griffin/FunctionParser.java

```diff
@@ -579,6 +579,9 @@ private Function checkAndCreateFunction(
                     .I$();
             Misc.freeObjList(args);
             throw SqlException.position(position).put("bad function factory (NULL), check log");
+        } else if (sqlExecutionContext.isDeterministic() && function.isNonDeterministic()) {
```

Comment:
nit: this doesn't look very clear with the current naming pattern.

Reply:
Improved in 245eb3b35c7cc5589c94c1384105ea28d71b7061